### PR TITLE
Add deprecation banner to old 3d panel

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
Added a banner to the 3D (Legacy) panel informing users that the panel is deprecated and has been replaced by the new 3D panel.

**Description**
Fixes https://github.com/foxglove/studio/issues/4422

image
image

### CODE REVIEW

**Review by jhurliman — APPROVED** [review]

**Comment by defunctzombie** [comment]
`This is a 3D (Legacy) panel. We are phasing out the 3D (Legacy) panel and recommend changing this panel to the 3D panel.`

**Comment by defunctzombie** [comment]
IIRC our plan is to merge this after we've made our blog announcements? The referenced issue notes that we would link to the blog post.

**Comment by 2metres** [comment]
Can we add a button that allows the user to swap to the new panel?

**Comment by defunctzombie** [comment]
`Your selected topics and settings will be lost.` maybe consider something like `You will need to re-configure your topics`. While it is true your topics and settings will be lost it is also true the user will be able to reproduce them by re-selecting them. This also makes me think that `Upgrade to the new 3D panel` is better as `Change this panel to the 3D panel`. Since "change panel" is closer to what happens if a user manually changes the panel. Settings are lost.

**Review by 2metres — APPROVED** [review]

**Comment by jtbandes** [comment]
@2metres please feel free to tweak appearance and wording (see slack discussion as well)

File: packages/studio-base/src/panels/ThreeDimensionalViz/Layout.tsx [hunk 1]
```diff
@@ -793,9 +814,58 @@ export default function Layout({
...
+  const [closedBanner, setClosedBanner] = useAppConfigurationValue<boolean>(
+    AppSetting.CLOSED_OLD3D_DEPRECATION_BANNER,
+  );
+  const [showUpgradeConfirmDialog, setShowUpgradeConfirmDialog] = useState(false);
+
+  const deprecationBanner =
+    closedBanner === true ? undefined : (
+      <Alert severity="info" color="warning" onClose={() => void setClosedBanner(true)}>
+        The 3D (Legacy) panel is now deprecated.{" "}
+        <Link onClick={() => setShowUpgradeConfirmDialog(true)}>Upgrade to the new 3D panel</Link>
```
Comment (2metres) [inline packages/studio-base/src/panels/ThreeDimensionalViz/Layout.tsx:826]:
```suggestion
        <Link color="inherit" onClick={() => setShowUpgradeConfirmDialog(true)}>Upgrade to the new 3D panel</Link>
```

File: packages/studio-base/src/panels/ThreeDimensionalViz/Layout.tsx [hunk 2]
```diff
@@ -793,9 +814,58 @@ export default function Layout({
...
+  const [closedBanner, setClosedBanner] = useAppConfigurationValue<boolean>(
+    AppSetting.CLOSED_OLD3D_DEPRECATION_BANNER,
+  );
+  const [showUpgradeConfirmDialog, setShowUpgradeConfirmDialog] = useState(false);
+
+  const deprecationBanner =
+    closedBanner === true ? undefined : (
+      <Alert severity="info" color="warning" onClose={() => void setClosedBanner(true)}>
+        The 3D (Legacy) panel is now deprecated.{" "}
+        <Link color="inherit" onClick={() => setShowUpgradeConfirmDialog(true)}>Upgrade to the new 3D panel</Link>
```
Comment (amacneil) [inline packages/studio-base/src/panels/ThreeDimensionalViz/Layout.tsx:826]:
```suggestion
        The 3D (Legacy) panel is deprecated.{" "}
        <Link color="inherit" onClick={() => setShowUpgradeConfirmDialog(true)}>Upgrade to the new 3D panel</Link>
        .
```

File: packages/studio-base/src/panels/ThreeDimensionalViz/Layout.tsx [hunk 3]
```diff
@@ -793,9 +814,58 @@ export default function Layout({
...
+      <Dialog open={showUpgradeConfirmDialog}>
+        <DialogTitle>Replace this panel?</DialogTitle>
+        <DialogContent>
+          <DialogContentText>
+            Are you sure you want to replace this 3D (Legacy) panel with the new 3D panel?
+            <br />
+            <br />
+            Your selected topics and settings will be lost.
+          </DialogContentText>
+        </DialogContent>
```
Comment (amacneil) [inline packages/studio-base/src/panels/ThreeDimensionalViz/Layout.tsx:842]:
```suggestion
        <DialogTitle>Upgrade to the new 3D panel</DialogTitle>
        <DialogContent>
          <DialogContentText>
            After upgrading this panel, you will need to reconfigure your selected topics.
          </DialogContentText>
        </DialogContent>
```

**Review by amacneil — APPROVED** [review]

**Comment by banisadr** [comment]
Happy to disagree and move forward here, but a delightful user experience would retain the topic information when you upgrade rather than require the user to reconfigure.

**Comment by amacneil** [comment]
Product review feedback:
- save selected topics and apply it to the new panel
- new wording is good

**Pull request merged**